b2share/modules/schemas/cli.py

```diff
@@ -46,3 +64,225 @@ def init(verbose):
+    comm = None
+    if community:
+        comm = get_community_by_name_or_id(community)
+    community_id = None
+    if comm:
+        community_id = comm.id
+    if verbose:
+        click.secho("filtering for community %s" % comm)
+    try:
+        block_schemas = BlockSchema.get_all_block_schemas(community_id=community_id)
+    except BlockSchemaDoesNotExistError:
+         raise click.ClickException("No block schemas found, community parameter was: %s" % community)
+    click.secho("BLOCK SCHEMA ID\t\t\t\tNAME\t\tMAINTAINER\tDEPRECATED\t#VERSIONS")
+    for bs in block_schemas:
+        bs_comm = Community.get(id=bs.community)
+        click.secho("%s\t%s\t%s\t%s\t\t%d" % (bs.id,bs.name[0:15].ljust(15), bs_comm.name[0:15].ljust(15), bs.deprecated, len(bs.versions)  ))
+        
+@schemas.command()
+@with_appcontext
+@click.option('-v','--verbose', is_flag=True, default=False)
+@click.option('-n','--name',help='set the name of the community')
+@click.option('-c','--community',help='set the maintaining community by name or id')
+@click.option('-d','--deprecated',help='(un)set deprecated bit, 1 is deprecated, 0 is not deprecated')
+@click.argument('block_schema_id')
+def block_schema_edit(verbose, name, community, deprecated, block_schema_id):
+    try:
+        val = UUID(block_schema_id, version=4)
+    except ValueError:
+        raise click.BadParameter("BLOCK_SCHEMA_ID is not a valid UUID (hexadecimal numbers and dashes e.g. fa52bec3-a847-4602-8af5-b8d41a5215bc )")
+    try:
+        block_schema = BlockSchema.get_block_schema(schema_id=block_schema_id)
+    except BlockSchemaDoesNotExistError:
+        raise click.BadParameter("No block_schema with id %s" % block_schema_id)
+    if not(name or community or deprecated):
+        raise click.ClickException("Noting to edit - at least one of name, community or deprecated must be provided.")
+    data = {}
+    if name:
+        data['name'] = name
+    if community:
+        comm = get_community_by_name_or_id(community)
+        if comm:
+            data['community'] = comm.id
+        else:
+            click.secho("Community not changed : no community exists with name or id: %s" % community)
+    if deprecated:
+        if not type(deprecated)==bool:
+            raise click.BadParameter("Deprecated should be True or False starting with a capital")
+        data['deprecated'] = deprecated
+    block_schema.update(data)
+    db.session.commit()
+        
+@schemas.command()
+@with_appcontext
+@click.option('-v','--verbose', is_flag=True, default=False)
+@click.argument('block_schema_id')
+@click.argument('json_file')
+def block_schema_create_version(verbose, block_schema_id, json_file):
+    """Assign a json-schema file conforming to the EUDAT block-schema part of the root-schema that defines metadata format for this block schema. Upon success, this creates a new version, the old versions are maintained and can be obtained by block_schema_list_versions subcommand."""
```

Comment:
Can we keep the 80 column restriction? This will make back-porting code to invenio easier.
In invenio we follow [PEP-8](https://www.python.org/dev/peps/pep-0008/#maximum-line-length) and PEP-257.